YAFCparser/Data/FactorioDataDeserializer_Context.cs

```diff
@@ -376,7 +376,7 @@ private class DataBucket<TKey, TValue>: IEqualityComparer<List<TValue>>
             private bool seal;
 
             // Replaces lists in storage with arrays. List with same contents get replaced with the same arrays
-            public void SealAndDeduplicate(TValue[] addExtra = null, IComparer<TValue> sort = null)
+            public void SealAndDeduplicate(TValue[] addExtra = null, IComparer<TValue> sort = null, System.Func<TKey, System.Func<TValue, bool>> filter = null)
             {
                 if (addExtra != null)
                     def = addExtra;
```

Comment:
There is something going on with the `def`ault value, which is used later down in `GetArray`. With the proposed changes there will be a different result depending on whether `storage `contains an empty list (`addExtra `is added in a filtered way in `SealAndDeduplicate`) or has no value at all (def/`addExtra` is used without the filter in `GetArray`).

Unfortunately I don't know the codebase good enough to understand immediately how this `DataBucket` thing is supposed to work.